### ISSUE TITLE
Add QueryChannelUpgradeProof or QueryChannelProof/QueryUpgradeProof

### ISSUE THREAD

**Issue by colin-axner** [cosmos/ibc-go#3722]
## Summary

Add a helper function to obtain proofs for testing, either via:
```go
proofChannel, proofUpgrade, proofHeight := endpoint.QueryChannelUpgradeProof()
```
or
```go 
proofChannel, proofHeight := endpoint.QueryChannelProof()
proofUpgrade, _ := endpoint.QueryUpgradeProof
```

These calls are used by:
- `endpoint.ChanUpgradeTry`
- `endpoint.ChanUpgradeAck`
- `endpoint.ChanUpgradeOpen`
- any test providing valid proofs of channels/upgrades (upgrade try, ack, open handlers and sub handlers (startFlushUpgradeHandshake)

A benefit of `QueryChannelProof` is that it can be reused for proofs used in the opening handshake

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

**Comment by DimitrisJim** [comment]
I believe the first approach is the most sensible here. Similar to this, `QueryConnectionHandshakeProof` also bundles all required `QueryProof` calls in a single place.

**Comment by damiannolan** [comment]
closed by #3731
